packages/svelte/types/index.d.ts

```diff
@@ -1991,6 +1991,22 @@ declare module 'svelte/transition' {
 }
 
 declare module 'svelte/events' {
+	/**
+	 * Attaches an event handler to the window and returns a function that removes the handler. Using this
+	 * rather than `addEventListener` will preserve the correct order relative to handlers added declaratively
+	 * (with attributes like `onclick`), which use event delegation for performance reasons
+	 *
+	 * */
+	export function on<Type extends keyof WindowEventMap>(window: Window, type: Type, handler: (this: Document, event: WindowEventMap[Type]) => any, options?: AddEventListenerOptions | undefined): () => void;
+
+	/**
+	 * Attaches an event handler to the document and returns a function that removes the handler. Using this
+	 * rather than `addEventListener` will preserve the correct order relative to handlers added declaratively
+	 * (with attributes like `onclick`), which use event delegation for performance reasons
+	 *
+	 * */
+	export function on<Type extends keyof DocumentEventMap>(document: Document, type: Type, handler: (this: Document, event: DocumentEventMap[Type]) => any, options?: AddEventListenerOptions | undefined): () => void;
+
```

Comment:
It seems like this adds the window and document events in the autocomplete of the second parameter regardless if the first parameter is `window` or `document`

https://tsplay.dev/wQQAVw

This can be quite confusing...is there a way to fix this behavior?

Reply:
I fear not - but since the squiggly tells you what's wrong immediately afterwards it should be fine.
This PR isn't quite right though because it adds these types to the outputs dts, I'm in the middle of fixing that